gensim/models/fasttext.py

```diff
@@ -807,6 +807,13 @@ def _load_fasttext_format(model_file, encoding='utf-8', full_model=True):
     with utils.open(model_file, 'rb') as fin:
         m = gensim.models._fasttext_bin.load(fin, encoding=encoding, full_model=full_model)
 
+    # Here we are checking for unsupported FB FT Modes
+    if m.loss != 1 and m.loss != 2:
```

Comment:
```suggestion
    if m.loss not in (1, 2):
```

Reply:
According to https://github.com/RaRe-Technologies/gensim/pull/3223, negative sampling is `loss=0`, not `loss=2`. Can you please double check? What's the correct value?

Reply:
Hi,
I checked it. In the parameters docstring in the models/_fasttext_bin, that I have copied here:
```
"""Holds data loaded from the Facebook binary.
Parameters

.
.
neg : int
    If non-zero, indicates that the model uses negative sampling.
loss : int
    If equal to 1, indicates that the model uses hierarchical sampling.
model : int
    If equal to 2, indicates that the model uses skip-grams.
"""
````
This suggests that `m.neg==0` means no negative sampling, and `m.neg == n` (where n is a positive integer) means negative sampling is being used. 
So I think if we are checking for negative sampling, the condition should be `if m.neg == 0` or `if m.neg != 0` based on what we want to do. From what I have read from the  #3223 I gather that negative sampling and hierarchical softmax are supported in loss. so we shouldn't really need to check for negative sampling i.e, `if m.neg != 0` (not really required)

I have made a list of what's supported and what's not based on #3223 . Please let me know if they are right, ill make changes as per that:

Supported:
a. Loss

    1. Negative Sampling
    2. Hierarchical Softmax
 b. Model 

    1. CBOW
    2. Skip-gram

Unsupported:
a. Loss:
 1. OneVsAll_Loss

b. Model:
  1. Supervised
Let me know if there are any inconsistencies in my understanding. I too want to help you complete this asap.

Reply:
I don't really understand that list. If the check is `== 0` or `!= 0`, why would we check for `== 1` and `== 2`?

I don't know the fasttext parameters, so can you please check the fasttext docs (not gensim) for what the parameters **actually** are? And then make changes to match that. Thanks.

Reply:
> Sorry, I don't really understand that list. If the check is `== 0` or `!= 0`, why would we check for `== 1` and `== 2`?

Hi yes you are right. We shouldn't be checking `==1` or `==2` in this case also if we are checking for negative sampling we must not, use `m.loss` at all we must check for `m.neg`. (Just to clarify the above ==1 and ==2 was not done by me, this PR was created by @Tewatia5355, Im just trying to figure out what are the losses and models not supported in fasttext. )
> 
> I don't know the fasttext parameters, so don't ask me :) Can you please check the fasttext docs or code (not gensim) for what the parameters **actually** are? And then make changes here to match that. Thanks.

Hi, yes sure I can investigate that and ill create a PR accordingly. Also if you can provide the link for fasttext docs that would be great. (just to be sure that I'm referring the right docs). Thanks!

Reply:
OK, please do. This PR seems incorrect so I'm removing the "Next release" tag – we need to fix it first, make the code correct.